src/GraphQLinq.Client/GraphQueryEnumerator.cs

```diff
@@ -63,62 +48,42 @@ private IEnumerable<T> DownloadData()
             }
 
             var enumerable = jObject[DataPathPropertyName][GraphQueryBuilder<T>.ResultAlias]
-                        .Select(token =>
-                        {
-                            var jsonSerializer = new JsonSerializer { ContractResolver = resolver };
-                            var jToken = queryType == QueryType.Collection ? token : token.Parent;
+                .Select(token =>
+                {
+                    var jsonSerializer = new JsonSerializer { ContractResolver = resolver };
+                    var jToken = queryType == QueryType.Collection ? token : token.Parent;
 
-                            if (mapper != null)
-                            {
-                                var result = jToken.ToObject<TSource>(jsonSerializer);
-                                return mapper.Invoke(result);
-                            }
+                    if (mapper != null)
+                    {
+                        var result = jToken.ToObject<TSource>(jsonSerializer);
+                        return mapper.Invoke(result);
+                    }
 
-                            return jToken.ToObject<T>(jsonSerializer);
-                        });
+                    return jToken.ToObject<T>(jsonSerializer);
+                });
 
             return enumerable;
         }
 
-        private string DownloadJson()
+        private async Task<string> DownloadJson(HttpClient httpClient)
         {
-            WebRequest.DefaultWebProxy.Credentials = CredentialCache.DefaultCredentials;
-
-            using (var webClient = new WebClient { Proxy = WebRequest.DefaultWebProxy })
+            try
             {
-                webClient.Headers.Add(HttpRequestHeader.ContentType, "application/json");
-
-                if (!string.IsNullOrEmpty(authorization))
-                {
-                    webClient.Headers.Add(HttpRequestHeader.Authorization, authorization);
-                }
-
-                try
-                {
-                    return webClient.UploadString(baseUrl, query);
-                }
-                catch (WebException exception)
-                {
-                    using (var responseStream = exception.Response?.GetResponseStream())
-                    {
-                        using (var streamReader = new StreamReader(responseStream))
-                        {
-                            return streamReader.ReadToEnd();
-                        }
-                    }
-                }
+                var content = new StringContent(query, Encoding.UTF8, "application/json");
+                var response = await httpClient.PostAsync("", content);
+                return await response.Content.ReadAsStringAsync();
```

Comment:
Instead of reading the whole response in a string it should return a stream and let the json serializer work with the stream.